File: library/src/main/java/me/proxer/library/enums/Language.java
package me.proxer.library.enums;

import com.serjltt.moshi.adapters.FallbackEnum;
import com.squareup.moshi.Json;

/**
 * An enum that holds the basic languages.
 *
 * @author Desnoo
 */
@FallbackEnum(name = "OTHER")
public enum Language {
    @Json(name = "de") GERMAN,
    @Json(name = "en") ENGLISH,
    @Json(name = "misc") OTHER,
}

File: library/src/main/kotlin/me/proxer/library/enums/Language.kt
package me.proxer.library.enums

import com.serjltt.moshi.adapters.FallbackEnum
import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

/**
 * An enum that holds the basic languages.
 *
 * @author Desnoo
 */
@JsonClass(generateAdapter = false)
@FallbackEnum(name = "OTHER")
enum class Language {

    @Json(name = "de")
    GERMAN,

    @Json(name = "en")
    ENGLISH,

    @Json(name = "misc")
    OTHER
}
